publish-to-maven-central/SDK4Mvn.aggr

```diff
@@ -37,37 +37,30 @@
+  <mavenMappings namePattern="org.apache.lucene.analyzers-smartcn" groupId="org.apache.lucene" artifactId="lucene-analyzers-smartcn"/>
+  <mavenMappings namePattern="(org\.apache\.felix)(\..+)" groupId="$1" artifactId="$1$2"/>
   <mavenMappings namePattern="org\.apache\.ant$" groupId="org.apache.ant" artifactId="ant"/>
-  <mavenMappings namePattern="(org.apache)\.(sshd)\.(core)" groupId="$1.$2" artifactId="$2-$3"/>
-  <mavenMappings namePattern="(org\.objectweb)\.([^.]+)$" groupId="org.ow2.asm" artifactId="$2"/>
-  <mavenMappings namePattern="(org\.objectweb)\.([^.]+)\.([^.]+)" groupId="org.ow2.asm" artifactId="$2-$3"/>
-  <mavenMappings namePattern="org.tukaani.xz" groupId="org.tukaani" artifactId="xz" versionPattern="([^.]+)\.([^.]+)\.0(?:\..*)?" versionTemplate="$1.$2"/>
-  <mavenMappings namePattern="org.hamcrest.core" groupId="org.hamcrest" artifactId="hamcrest-core"/>
+  <mavenMappings namePattern="(org.apache)\.(sshd)\.(core|osgi)" groupId="$1.$2" artifactId="$2-$3"/>
+  <mavenMappings namePattern="(org\.bouncycastle)\.([^.]+)" groupId="$1" artifactId="$2-jdk15on" versionPattern="([^.]+)\.([^.]+)\.0(?:\..*)?" versionTemplate="$1.$2"/>
   <mavenMappings namePattern="(org\.junit)\.([^.]+)\.([^.]+)" groupId="$1.$2" artifactId="junit-$2-$3"/>
   <mavenMappings namePattern="(org\.junit)\.([^.]+)\.([^.]+)\.([^.]+)" groupId="$1.$2" artifactId="junit-$2-$3-$4"/>
   <mavenMappings namePattern="(org)\.(opentest4j)" groupId="$1.$2" artifactId="$2"/>
   <mavenMappings namePattern="(org)\.(apiguardian)" groupId="$1.$2" artifactId="$2-api"/>
+  <mavenMappings namePattern="org.apache.batik.([^.]+)" groupId="org.apache.xmlgraphics" artifactId="batik-$1" versionPattern="([^.]+)\.([^.]+)\.0(?:\..*)?" versionTemplate="$1.$2"/>
   <mavenMappings namePattern="org.junit" groupId="junit" artifactId="junit" versionPattern="([^.]+)\.([^.]+)\.0(?:\..*)?" versionTemplate="$1.$2"/>
   <mavenMappings namePattern="org.w3c.css.sac" groupId="org.eclipse.birt.runtime" artifactId="org.w3c.css.sac"/>
-  <mavenMappings namePattern="org.apache.batik.css" groupId="org.apache.xmlgraphics" artifactId="batik-css" versionPattern="([^.]+)\.([^.]+)\.0(?:\..*)?" versionTemplate="$1.$2"/>
-  <mavenMappings namePattern="com.google.gson" groupId="com.google.code.gson" artifactId="gson"/>
-  <mavenMappings namePattern="com.sun.jna.platform" groupId="net.java.dev.jna" artifactId="jna-platform"/>
-  <mavenMappings namePattern="com.sun.jna" groupId="net.java.dev.jna" artifactId="jna"/>
-  <mavenMappings namePattern="org.w3c.dom.events" groupId="org.eclipse.orbit.bundles" artifactId="org.w3c.dom.events"/>
-  <mavenMappings namePattern="org.w3c.dom.smil" groupId="org.eclipse.orbit.bundles" artifactId="org.w3c.dom.smil"/>
-  <mavenMappings namePattern="org.w3c.dom.svg" groupId="org.eclipse.orbit.bundles" artifactId="org.w3c.dom.svg"/>
+  <mavenMappings namePattern=".*" groupId="\$maven-groupId\$" artifactId="\$maven-artifactId\$" versionPattern=".*" versionTemplate="\$maven-version\$"/>
```

Comment:
I asked via email for your feedback in here but none was forthcoming:

https://github.com/eclipse-cbi/p2repo-aggregator/issues/11

I think we/you can live with this explicit opt-in rule.  I want to offer that flexibility to ensure that new behavior is opt-in rather than "we think this is best for you so you get it whether you want it or not."

Reply:
> "we think this is best for you so you get it whether you want it or not."

IMO, this usually results in better results. More configuration is more complexity, more room for errors; it is often less productive. I think this is the case here, and I don't see addition of this rule as an added-value over the current state (where this rule is hardcoded as default when possible) but more as an extra complication that doesn't make maintenance of this .aggr file, nor of the aggregator itself, easier.

Reply:
Sometimes opinions differ.  This option does not and will not require further/future maintenance.